src/socket/address.cr

```diff
@@ -100,21 +101,43 @@ class Socket
     # Raises `Socket::Error` if *address* does not contain a valid IP address or
     # the port number is out of range.
     #
+    # Scoped/Zoned IPv6 link-local addresses are supported per RFC4007, e.g.
+    # `fe80::abcd%eth0`.
+    #
     # ```
     # require "socket"
     #
     # Socket::IPAddress.new("127.0.0.1", 8080)                 # => Socket::IPAddress(127.0.0.1:8080)
     # Socket::IPAddress.new("fe80::2ab2:bdff:fe59:8e2c", 1234) # => Socket::IPAddress([fe80::2ab2:bdff:fe59:8e2c]:1234)
+    # Socket::IPAddress.new("fe80::4567:8:9%eth0", 443)        # => Socket::IPAddress([fe80::4567:8:9]:443)
```

Comment:
```suggestion
    # Socket::IPAddress.new("fe80::4567:8:9%eth0", 443)        # => Socket::IPAddress([fe80::4567:8:9%eth0]:443)
```

Reply:
Fixed and clarified the RFC4007 section in the comment as well.